coderd/userauth.go

```diff
@@ -1096,7 +1096,7 @@ func (api *API) userOAuth2Github(rw http.ResponseWriter, r *http.Request) {
 	}
 	// If the user is logging in with github.com we update their associated
 	// GitHub user ID to the new one.
-	if externalauth.IsGithubDotComURL(api.GithubOAuth2Config.AuthCodeURL("")) && user.GithubComUserID.Int64 != ghUser.GetID() {
+	if externalauth.IsGithubDotComURL(api.GithubOAuth2Config.OAuth2Config.AuthCodeURL("")) && user.GithubComUserID.Int64 != ghUser.GetID() {
```

Comment:
Nit: This needs an explanation why we're bypassing the default `AuthCodeURL` here.

(An even better option may be to refactor `promoauth` so that `OAuth2Config` has a `domain` method or some such. Not a requirement though.)